### PR TITLE
[Merged by Bors] - feat: port RingTheory.Subring.Basic

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/Subring/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,1485 @@
...
+instance (priority := 75) toCommRing {R} [CommRing R] [SetLike S R] [SubringClass S R] :
+    CommRing s :=
+  Subtype.coe_injective.commRing (↑) rfl rfl (fun _ _ => rfl) (fun _ _ => rfl) (fun _ => rfl)
+    (fun _ _ => rfl) (fun _ _ => rfl) (fun _ _ => rfl) (fun _ _ => rfl) (fun _ => rfl) fun _ => rfl
+#align subring_class.to_comm_ring SubringClass.toCommRing
+
+-- Prefer subclasses of `ring` over subclasses of `subring_class`.
+/-- A subring of a domain is a domain. -/
+instance (priority := 75) {R} [Ring R] [IsDomain R] [SetLike S R] [SubringClass S R] : IsDomain s :=
+  NoZeroDivisors.to_isDomain _
```
Comment (Vierkantor) [inline Mathlib/RingTheory/Subring/Basic.lean:119]: This seems to time out over the unification problem `Semiring.toNonUnitalSemiring.1 =?= (SubsemiringClass.toNonAssocSemiring s).1`. I blame leanprover/lean4#2003 :(

Comment (jcommelin) [reply Mathlib/RingTheory/Subring/Basic.lean]: 2003 is merged :octopus:

**Review by jcommelin — APPROVED** [review]

**Comment by jcommelin** [comment]
Recorded an issue over at #2210. We should debug this. But it doesn't have to block the port of this file.

bors merge

**Comment by eric-wieser** [comment]
This seems to have broken mathport, https://github.com/leanprover-community/mathport/issues/228

**Pull request closed without merging**